diffpy/pdfmorph/pdfmorph_api.py

```diff
@@ -0,0 +1,200 @@
+    # morphing (xobj, yobj) pair to (xref, yref) pair with scaling
+    from diffpy.pdfmorph import pdfmorph, default_config, plot_morph
+
+    morph_cfg = dict(default_config)
+    morph_cfg['scale'] = 1.01
+    rv = pdfmorph(xobj, yobj, xref, yref, **morph_cfg)
+    morph_chain, morphed_cfg, rw, pearson = rv
+
+    plot_morph(morph_chain)
+    print(morphed_cfg, rw)
+    ..
+    """
+    operation_dict = {}
+    refpars = []
+    # base config
+    rv_cfg = dict(default_config)
+    if kwargs:
+        rv_cfg.update(kwargs)
+    # configure morph operations
+    active_morphs = [k for k, v in rv_cfg.items() if (v is not None) and k in
+                     morph_step_dict]
+    rv_cfg['rmin'] = rmin
+    rv_cfg['rmax'] = rmax
+    rv_cfg['rstep'] = rstep
+    # configure smear, guess baselineslope when it is not provided
+    if (rv_cfg.get('smear') is not None
+        and rv_cfg.get('baselineslope') is None):
+        rv_cfg['baselineslope'] = -0.5
+    # config dict defines initial guess of parameters
+    chain = morphs.MorphChain(rv_cfg)
+    # rgrid
+    chain.append(morphs.MorphRGrid())
+    # configure morph chain
+    for k in active_morphs:
+        morph_cls = morph_step_dict[k]
+        if k == 'smear':
+            [chain.append(el()) for el in morph_cls]
+            refpars.append('baselineslope')
+        else:
+            chain.append(morph_cls())
+        refpars.append(k)
+    # exclude fixed options
+    if fixed_operations:
+        if not isinstance(fixed_operations, Iterable):
+            fixed_operations = [fixed_operations]
+        for opt in fixed_operations:
+            refpars.remove(opt)
+        print("== INFO: Following morphing steps are fixed ==:\n{}\n"
+              .format('\n'.join(fixed_operations)))
+    # define refiner
+    refiner = ref.Refiner(chain, xobj, yobj, xref, yref)
+    if pearson:
+        refiner.residual = refiner._pearson
+    if addpearson:
+       refiner.residual = refiner._addpearson
+    # execute morphing
+    if refpars and refine:
+        # This works better when we adjust scale and smear first.
+        if "smear" in refpars:
+            rptemp = ["smear"]
+            if "scale" in refpars:
+                rptemp.append("scale")
+            refiner.refine(*rptemp)
+        # Refine all params
+        refiner.refine(*refpars)
+    else:
+        # no operation if refine=False or refpars is empty list
+        chain(xobj, yobj, xref, yref)
+
+    # summary
+    rw = tools.getRw(chain)
+    pcc = tools.getPearson(chain)
+    # restore rgrid
+    chain[0] = morphs.Morph()
+    chain(xobj, yobj, xref, yref)
+    # print output
+    if verbose:
+        output = "\n".join(["# {} = {:.6f}".format(k, v) for k, v in \
+                rv_cfg.items() if v is not None])
+        output += "\n# Rw = {:.6f}".format(rw)
+        output += "\n# Pearson = {:.6f}".format(pcc)
+        print(output)
+
+    return chain, rv_cfg, rw, pcc
+
+
+def plot_morph(chain, fig=None, **kwargs):
```

Comment:
It seems that this function updates only the `ax` axis object.
A better interface would be to have `plot_morph(chain, ax=None)` so that
user can plot to any specific Axis, which is however quite clunky now.
Creating `ax` with a `subplots(**kwargs)` call is not a business for `plot_morph` to do,
I would suggest to use `ax = pyplot.gca() if ax is None else ax`.

I would also return a list of created Line2D objects, which could be then conveniently formatted.
User already has the `fig` object or can get it as `ax.figure`.